### PR TITLE
Add a reference Horovod Dockerfile

### PR DESCRIPTION
**Pull request by alsrgv**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,82 @@
+FROM nvidia/cuda:8.0-cudnn6-devel-ubuntu16.04
+
+# TensorFlow version is tightly coupled to CUDA and cuDNN so it should be selected carefully
+ENV TENSORFLOW_VERSION=1.4.0
+
+# Python 2.7 or 3.5 are supported by Ubuntu Xenial out of the box
```
Comment (sblotner) [inline Dockerfile]: are --> is

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+# Set default NCCL parameters
+RUN echo NCCL_DEBUG=INFO >> /etc/nccl.conf && \
+    echo NCCL_SOCKET_IFNAME=^docker0 >> /etc/nccl.conf
+
+# Install OpenSSH for MPI to communicate between containers
+RUN apt-get install -y --no-install-recommends openssh-client openssh-server && \
+    mkdir -p /var/run/sshd
+
+# Allow OpenSSH to talk to each other without asking for confirmation
```
Comment (sblotner) [inline Dockerfile]: maybe change each other --> containers

File: docs/docker.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+## Horovod in Docker
+
+To ease the installation process on GPU machines, we have published the reference [Dockerfile](../Dockerfile) that
+allows you to get started with Horovod in minutes. The container will include [Examples](../examples) in the `/examples`
```
Comment (sblotner) [inline docs/docker.md]: will include --> includes

File: docs/docker.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+directory for a convenient quickstart.
```
Comment (sblotner) [inline docs/docker.md]: remove "for a convenient quickstart"

File: docs/docker.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+## Horovod in Docker
+
+To ease the installation process on GPU machines, we have published the reference [Dockerfile](../Dockerfile) that
```
Comment (sblotner) [inline docs/docker.md]: ease --> streamline
"we have published the reference [Dockerfile](../Dockerfile) that" --> "we published the reference [Dockerfile](../Dockerfile) so you can get started with Horovod in minutes."

File: docs/docker.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+### Building
+
+```bash
+$ mkdir horovod-docker
+$ wget -O horovod-docker/Dockerfile https://raw.githubusercontent.com/uber/horovod/master/Dockerfile
+$ docker build -t horovod:latest horovod-docker
+```
+
+Before building, you can modify `Dockerfile` to your liking, e.g. select different CUDA, TensorFlow or Python version.
```
Comment (sblotner) [inline docs/docker.md]: select different --> select a different
add a comma after TensorFlow

Comment (sblotner) [reply docs/docker.md]: consider moving this sentence right below the Building heading

File: docs/docker.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+### Running on a single machine
+
+After the container is built, you can run it using [nvidia-docker](https://github.com/NVIDIA/nvidia-docker).
```
Comment (sblotner) [inline docs/docker.md]: remove "you can"

File: docs/docker.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+```bash
+$ nvidia-docker run -it horovod:latest
+root@c278c88dd552:/examples# mpirun -np 4 -H localhost:4 python keras_mnist_advances.py
+```
+
+You may notice that this command does not have few options recommended in other parts of documentation:
```
Comment (sblotner) [inline docs/docker.md]: have few --> have a few

File: docs/docker.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+After the container is built, you can run it using [nvidia-docker](https://github.com/NVIDIA/nvidia-docker).
+
+```bash
+$ nvidia-docker run -it horovod:latest
+root@c278c88dd552:/examples# mpirun -np 4 -H localhost:4 python keras_mnist_advances.py
+```
+
+You may notice that this command does not have few options recommended in other parts of documentation: 
+`-bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH`.  These options are already set by default in the
+docker container, and so are not required to repeated in the command.
```
Comment (sblotner) [inline docs/docker.md]: docker --> Docker
", and so are not required to repeated in the command." --> "so you don't need to repeat them in the command."

File: docs/docker.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+If you don't run your container in privileged mode, you may see the following message:
+
+```
+[a8c9914754d2:00040] Read -1, expected 131072, errno = 1
+```
+
+This message can be safely ignored.
```
Comment (sblotner) [inline docs/docker.md]: Consider rephrasing to "You can ignore this message."

File: docs/docker.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+### Running on multiple machines
+
+Here we describe a simple way involving a shared filesystem `/mnt/share` and using a common port number `12345` for SSH
```
Comment (sblotner) [inline docs/docker.md]: "way" --> "example"
"and using" --> "using"
"for SSH" --> "for the SSH"

File: docs/docker.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+daemon that will be run on all the containers. `/mnt/share/ssh` would contain typical `id_rsa` and `authorized_keys`
```
Comment (sblotner) [inline docs/docker.md]: contain typical --> contain a typical

Comment (alsrgv) [reply docs/docker.md]: Thanks!

**Pull request merged**